api/v1alpha1/httproute_types.go

```diff
@@ -34,11 +34,23 @@ type HTTPRouteSpec struct {
 
 // HTTPRouteHost is the configuration for a given host.
 type HTTPRouteHost struct {
-	// Hostnames are a list of hosts names that match this host
-	// block.
+	// Hostname is the fully qualified domain name of a network host,
+	// as defined by RFC 3986. Note the following deviations from the
+	// "host" part of the URI as defined in the RFC:
 	//
-	// TODO: RFC link
-	Hostnames []string `json:"hostnames"`
+	// 1. IPs are not allowed.
```

Comment:
Would it be simpler to say that the value must be a `reg-name` per RFC 3986?

Reply:
I think stating "IPs are not allowed" will make sense to most, if not all, users. I just spent a few minutes reviewing https://tools.ietf.org/html/rfc3986 and I still don't understand what a `reg-name` is. Maybe we can iterate on the godoc for this field to include `reg-name`?

Reply:
It is useful to state explicitly that IP addresses are prohibited, fair enough.  I think referencing the specific syntax item is also useful.  From the RFC:

     host          = IP-literal / IPv4address / reg-name
     reg-name      = *( unreserved / pct-encoded / sub-delims )
     pct-encoded   = "%" HEXDIG HEXDIG
     unreserved    = ALPHA / DIGIT / "-" / "." / "_" / "~"
     sub-delims    = "!" / "$" / "&" / "'" / "(" / ")" / "*" / "+" / "," / ";" / "="